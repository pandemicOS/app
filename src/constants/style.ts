export const STYLE = {
  FONT: {
    SIZE: {
      HEADING_BIG: 32,
      HEADING_1: 23,
      HEADING_2: 17,
      BUTTON_LARGE: 17,
      PARAGRAPH: 15,
      PARAGRAPH_SMALL: 13
    },
    WEIGHTS: {
      LIGHT: '300' as any,
      REGULAR: '400' as any,
      MEDIUM: '600' as any,
      BOLD: '700' as any
    }
  },
  COLOR: {
    PRIMARY: '#FFBB00',
    SECONDARY: '#654400',
    TERTIARY: '#7C7C7C',
    QUATERNARY: '#4A90E2',
    QUINARY: '#F6F6F6',
    SENARY: '#646565',
    SEPTENARY: '#1ED66C',
    OCTANARY: '#D5DEF1',
    NONARY: '#74B4FF',
    DENARY: '#3A93FB',
    BLACK: '#1E272E',
    WHITE: '#FFF',
    WHITE_SMOKE: '#FAFAFA',
    LIGHT_GRAY: '#D2D2D2',
    SHADOW_GRAY: '#D7D7D7',
    DIM_GRAY: '#676767',
    DARK_GRAY: '#898989',
    LIGHTEST_GRAY: '#E7E7E7',
    MID_GRAY: '#6A6A77',
    ALMOST_BLACK: '#303030',
    RED_CARDINAL: '#C70D32',
    DARK_RED: '#FF0000',
    CURIOUS_BLUE: '#4191CE',
    INVISIBLE: 'rgba(0, 0, 0, 0)',
    INVISIBLE_WHITE: 'rgba(255, 255, 255, 0)',
    GREEN: '#7ED321',
    RED: '#CE384A',
  },
  BORDER: {
    GLOBAL_RADIUS: 4,
    QUARTER: 25
  }
};
