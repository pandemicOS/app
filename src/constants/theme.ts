import { Dimensions, StyleSheet } from 'react-native';
import { STYLE } from './style';

export const CONTAINER_MAX_WIDTH = 600;
const screenSizeW = Dimensions.get('window').width;
const screenPadding =
  screenSizeW >= CONTAINER_MAX_WIDTH
    ? (screenSizeW - CONTAINER_MAX_WIDTH) / 2
    : 40;

// Generic navigation properties
const navigation = {
  elevation: 0,
  shadowOpacity: 0,
  shadowOffset: {
    height: 0,
    width: 0
  },
  shadowRadius: 0,
  shadowColor: 'transparent',
  borderBottomWidth: 0
};

export const THEME = {
  screenPadding: {
    paddingHorizontal: screenPadding
  },
  navHeader: {
    backgroundColor: STYLE.COLOR.WHITE,
    height: 60,
    borderBottomWidth: 0,
    elevation: 0
  },
  backButton: {
    marginLeft: 25,
    width: 40,
    height: 40,
    justifyContent: 'center'
  },
  menuButton: {
    width: 40,
    height: 40,
    marginLeft: 20,
    justifyContent: 'center'
  },
  navHeaderTitleText: {
    fontSize: 16,
    flex: 1,
    color: STYLE.COLOR.BLACK,
    textAlign: 'center',
    alignSelf: 'center',
  },
  navHeaderGray: {
    ...navigation,
    backgroundColor: STYLE.COLOR.WHITE
  },
  navHeaderWhite: {
    ...navigation,
    backgroundColor: STYLE.COLOR.WHITE
  },
  navHeaderButton: {
    padding: 10
  },
  navLogo: {
    height: 29,
    width: 134,
    marginRight: 20
  },
  regularText: {
    color: STYLE.COLOR.DIM_GRAY,
    fontWeight: STYLE.FONT.WEIGHTS.REGULAR,
  },
  boldText: {
    color: STYLE.COLOR.DIM_GRAY,
    fontWeight: STYLE.FONT.WEIGHTS.BOLD,
  },
  titleText: {
    marginBottom: 20,
    textAlign: 'center',
    fontSize: 20,
    fontWeight: STYLE.FONT.WEIGHTS.BOLD,
    color: STYLE.COLOR.CURIOUS_BLUE
  },
  textError: {
    color: STYLE.COLOR.RED_CARDINAL,
    backgroundColor: 'transparent'
  },
  errorWrapper: {
    marginTop: 8,
    marginBottom: 16
  }
} as any;
