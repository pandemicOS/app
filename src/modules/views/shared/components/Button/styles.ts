import { StyleSheet } from 'react-native';

import { STYLE } from '../../../../../constants';

export default StyleSheet.create({
  container: {
    justifyContent: 'center',
    backgroundColor: STYLE.COLOR.PRIMARY,
    borderRadius: 4,
    height: 55,
  },
  wrapper: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center'
  },
  textWrapper: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  text: {
    fontSize: 16,
    color: STYLE.COLOR.BLACK,
    paddingLeft: 12,
    paddingRight: 5,
    fontWeight: STYLE.FONT.WEIGHTS.MEDIUM,
  },
  secondaryText: {
    fontSize: 12,
    marginTop: 1,
    paddingRight: 5,
    color: STYLE.COLOR.TERTIARY,
    fontWeight: STYLE.FONT.WEIGHTS.MEDIUM,
  }
});
