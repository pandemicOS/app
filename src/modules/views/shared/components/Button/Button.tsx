import React from 'react';
import { View, Text, TouchableOpacity, ActivityIndicator, StyleProp, ViewStyle, TextStyle } from 'react-native';

import styles from './styles';
import { STYLE } from '../../../../../constants';

export interface IButtonProps {
  iconStyle?: number;
  text: string;
  secondaryText?: string;
  isLoading?: boolean;
  isDisabled?: boolean;
  isSelected?: boolean;
  onPress?: () => void;
  style?: StyleProp<ViewStyle>;
  textStyle?: StyleProp<TextStyle>;
  secondaryTextStyle?: StyleProp<TextStyle>;
  imageTintColor?: string;
}

export default class Button extends React.PureComponent<IButtonProps> {
  public render() {
    const { text, secondaryText, isDisabled, isSelected, isLoading, onPress, style, textStyle, secondaryTextStyle } = this.props;
    return (
      <TouchableOpacity
        style={[styles.container, style, isDisabled ? { opacity: 0.7 } : {}, isSelected ? { backgroundColor: STYLE.COLOR.SECONDARY } : {}]}
        onPress={onPress}
        disabled={isLoading || isDisabled}
      >
        {isLoading ? (
          <ActivityIndicator animating={true} color={STYLE.COLOR.DIM_GRAY} />
        ) : (
          <View style={styles.wrapper}>
            <View style={styles.textWrapper}>
              <Text style={[styles.text, textStyle]}>{text}</Text>
              <Text style={[styles.secondaryText, secondaryTextStyle]}>{secondaryText}</Text>
            </View>
          </View>
        )}
      </TouchableOpacity>
    );
  }
}
