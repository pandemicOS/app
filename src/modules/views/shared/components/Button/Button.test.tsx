import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import Button, { IButtonProps } from './Button';
import { IMAGES } from '../../../../../constants';

describe('Button', () => {
  let props: IButtonProps;
  beforeEach(() => {
    props = {
      icon: '',
      text: '',
      isLoading: false,
      onPress: jest.fn(),
    };
  });
  it('renders without crashing', () => {
    const rendered = renderer.create(<Button {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('renders without crashing with ActivityIndicator', () => {
    props.isLoading = true;
    const rendered = renderer.create(<Button {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should handle onPress', async () => {
    const wrapper = shallow(<Button {...props} />);
    await (wrapper.find('TouchableOpacity').first().prop('onPress') as any)();
    expect(props.onPress).toBeCalled();
  });

  it('should render without onPress', async () => {
    props.onPress = undefined;
    const rendered = renderer.create(<Button {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });

  it('should render with icon', async () => {
    props.icon = 'LOGIN';
    const rendered = renderer.create(<Button {...props} />).toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
