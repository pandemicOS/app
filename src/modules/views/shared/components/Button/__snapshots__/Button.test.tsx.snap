// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders without crashing 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "borderRadius": 4,
      "height": 55,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 12,
              "paddingRight": 5,
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#7C7C7C",
              "fontSize": 12,
              "fontWeight": "600",
              "marginTop": 1,
              "paddingRight": 5,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Button renders without crashing with ActivityIndicator 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "borderRadius": 4,
      "height": 55,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#676767"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;

exports[`Button should render with icon 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "borderRadius": 4,
      "height": 55,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 12,
              "paddingRight": 5,
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#7C7C7C",
              "fontSize": 12,
              "fontWeight": "600",
              "marginTop": 1,
              "paddingRight": 5,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`Button should render without onPress 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#4A90E2",
      "borderRadius": 4,
      "height": 55,
      "justifyContent": "center",
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFF",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 12,
              "paddingRight": 5,
            },
            undefined,
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#7C7C7C",
              "fontSize": 12,
              "fontWeight": "600",
              "marginTop": 1,
              "paddingRight": 5,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
